datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -1554,7 +1554,8 @@ impl HashJoinStream {
                     }
                     Some(result.map(|x| x.0))
                 }
-                other => {
+                Some(err) => Some(err),
```

Comment:
👍 this is also found in the refactor pr https://github.com/apache/arrow-datafusion/pull/4377/files#